pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -137,6 +151,16 @@ func (rc *reconciler) reconcile() {
 	for _, attachedVolume := range rc.actualStateOfWorld.GetAttachedVolumes() {
 		if !rc.desiredStateOfWorld.VolumeExists(
 			attachedVolume.VolumeName, attachedVolume.NodeName) {
+
+			// Don't even try to start an operation if there is already one running
+			// This check must be done before we do any other checks, as otherwise the other checks
+			// may pass while at the same time the volume leaves the pending state, resulting in
+			// double detach attempts
+			if rc.attacherDetacher.IsOperationPending(attachedVolume.VolumeName, "") {
```

Comment:
Because there is a race condition otherwise where it may happen that a volume is tried to be detached again when the previous detach finishes right before we call the operation executor. It's really just a very short time frame where this may happen, but I encountered this in one of my test cases, resulting in non deterministic number of calls to the detacher, making the tests flaky.